src/components/common/Balance.js

```diff
@@ -8,24 +8,35 @@ const CustomDiv = styled(List)`
     display: inline;
 `
 
+const FRAC_DIGITS = 5
+
 const Balance = ({ amount }) => {
     if (!amount) {
         throw new Error("amount property should not be null")
     }
     let amountShow = formatNEAR(amount)
+
     return (
-        <CustomDiv>
+        <CustomDiv title={['0','0.0001'].includes(utils.format.formatNearAmount(amount, (FRAC_DIGITS || 1) - 1)) ? `${formatWithCommas(amount)} yoctoⓃ` : ``}>
```

Comment:
Can we break up this long line? Maybe something like this:

```jsx
<CustomDiv title={
  ['0', '0.0001'].includes(
    utils.format.formatNearAmount(amount, (FRAC_DIGITS || 1) - 1)
  )
    ? `${formatWithCommas(amount)} yoctoⓃ`
    : ``
}>
```

I find that much easier to look at!

And now that I formatted it like that, I have some other questions:

* `FRAC_DIGITS || 1` – we know that `FRAC_DIGITS` is always `5`, so the `|| 1` will never be evaluated
* is there value in saying `FRAC_DIGITS - 1` instead of `4`?
* `['0', '0.0001']` seems like a very precise list – is this generic enough to catch all edge cases we're interested in catching? 
* do we need our own `formatWithCommas`? It seems like maybe `formatNearAmount` or `parseNearAmount` can do this for us: https://github.com/nearprotocol/nearlib/blob/master/test/utils/format.test.js#L7
* using `formatNearAmount` and then comparing the result to an array of strings seems like a roundabout way to check the magnitude of a number – can we directly use `<` instead? `amount <= 10 ** 21` – would this be correct?

Reply:
I think it's best to extract complex logic into another helper function vs have it inline in JSX.

Also I am nor sure why set `FRAC_DIGITS` in a way where you always have to subtract `1`.

Reply:
> * `FRAC_DIGITS || 1` – we know that `FRAC_DIGITS` is always `5`, so the `|| 1` will never be evaluated

It's added in case `FRAC_DIGITS` will be set to `0`

> * is there value in saying `FRAC_DIGITS - 1` instead of `4`?

Yes, so that it can be controlled by `FRAC_DIGITS` only. From issue description, I understand that we want to show the amount in `yoctoN` in a situation where fractional digits number is one less than we are showing. So if we round it to `0.00001` we want to show it for amounts `<=0.0001`. Or did I misunderstood @vgrichina?

> * `['0', '0.0001']` seems like a very precise list – is this generic enough to catch all edge cases we're interested in catching?

Yes, for `FRAC_DIGITS = 5`. But `0.0001` needs to be changed to handle other `FRAC_DIGITS` values. Thanks for pointing that out.

> * do we need our own `formatWithCommas`? It seems like maybe `formatNearAmount` or `parseNearAmount` can do this for us:

`formatNearAmount` divides given amount by 24
https://github.com/nearprotocol/nearlib/blob/master/src.ts/utils/format.ts#L27
`parseNearAmount` multiplies given amount by 24
https://github.com/nearprotocol/nearlib/blob/master/src.ts/utils/format.ts#L51

> * using `formatNearAmount` and then comparing the result to an array of strings seems like a roundabout way to check the magnitude of a number – can we directly use `<` instead? `amount <= 10 ** 21` – would this be correct?

We are operating on strings there, so I wanted it to be consistent.

> Also I am nor sure why set `FRAC_DIGITS` in a way where you always have to subtract `1`.

In `title` tag it's because we want to show title if the amount is `<=0.0001`, so it's one fractional digit less. In `formatNEAR` it's because It's adding `1` at the end.

Reply:
I think we need to create `utils.format.formatYoctoNearAmount` and use it here.

Also I feel like less hacky way to compare for threshold would be to use `BN` for math and compare against specific threshold number. So something like:

```
const YOCTO_NEAR_THRESHOLD = new BN('10', 10).pow(new BN(NEAR_NOMINATION_EXP - FRAC_DIGITS +1, 10));

// ... later in method
const amount = new BN(amountStr);
if (amount.lt(YOCTO_NEAR_THRESHOLD)) {
   return formatYoctoNearAmount(amount) + ' yoctoⓃ';
}
return  formatNearAmount(amount) + ' Ⓝ';
```

also likely best to use https://en.wikipedia.org/wiki/Non-breaking_space instead of regular space here

Reply:
Done. `formatWithCommas` is copied from nearlib utils, it can be removed if we can export `formatWithCommas` from nearlib.